source/TextBlade.ConsoleRunner/Game.cs

```diff
@@ -89,15 +92,15 @@ private void ApplyResultsIfBattle(ICommand command)
         var dungeon = _currentLocation as Dungeon;
 
         // Kinda a special case for battle commands
-        if (!(command is IBattleCommand battleCommand))
+        if (command is not IBattleCommand battleCommand)
         {
             return;
         }
 
         if (battleCommand.IsVictory)
         {
             // Wipe out the dungeon floor's inhabitants.
-            dungeon.OnVictory(_inventory);
+            dungeon?.OnVictory(_inventory);
```

Comment:
`dungeon` should never be null here, because `IBattleCommand` guarantees that it should be set. Does it make sense to add a `?` here? Or should we do the usual if-null-throw to satisfy the warning?

Reply:
IBattleCommand only guarantee that _currentLocation implements the interface, not that it descends from Dungeon. If a Location implements IBatteCeommands but does not descend from Dungeon, it will be null